src/chapter_2/main.mo

```diff
@@ -16,6 +16,9 @@ actor chapter2{
   let members = HashMap.HashMap<Principal, Member>(1, Principal.equal, Principal.hash);
   /// function to add new member and give error if already exists
   public shared ({caller}) func addMember(member: Member) : async Result<(), Text>{
+    if (member.age < 1 or member.age > 100) {
+      return #err("Age must be between 1 and 100");
+    };
```

Comment:
right now if the caller already exists it will return this error. Please make sure that this error is only returned if the member does not exist